zerver/lib/bugdown/__init__.py

```diff
@@ -1415,14 +1415,21 @@ def run(self, lines: List[str]) -> List[str]:
             is_next_item = (m and current_list
                             and current_indent == len(m.group(1)) // self.TAB_LENGTH)
 
-            if not is_next_item:
+            is_blank_line = line.strip() == ""
+
+            if not is_next_item and not is_blank_line:
                 # There is no more items in the list we were processing
                 new_lines.extend(self.renumber(current_list))
                 current_list = []
 
             if not m:
-                # Ordinary line
-                new_lines.append(line)
+                # If current list still has items and current line
+                # is a blank line then do not push it to new line,
+                # otherwise push it to the new line regardless of
+                # whether line is blank line or ordinary line
+                if not current_list:
+                    # Ordinary line
+                    new_lines.append(line)
```

Comment:
Can you include a test case where this comes up in your unit tests?

Reply:
@timabbott Sure will add. Thanks!

Reply:
Can you explain precisely what situation this block comes up in?

Reply:
If we have a list like this (adde - to avoid renumbering) :  
-1. A

-1. B

-1. C

-1. D
som ordinary text
-1. AA

-1. AA

It will renumber first list before the paragraph as list 1 and renumber list after pararaph as list 2

Reply:
just to show that normal code block, paragraph etc. acts as a list separator unlike blankline.

Reply:
When it reaches `some ordinary text` by that time `current_list` should have gotten empty, renumbered and added to the `new_lines` because `is_next_item` is false and `is_blank_line` is false.  Then, in that case, we should add this normal paragraph in `new_lines`.

Reply:
If any blank line is encountered then `current_list` will not be emptied because this condition will be false `if not is_next_item and not is_blank_line:`. In that case it'll reach this block and we'll not add blank line in `new_lines` since their might be more list items.